code/game/objects/items/weapons/cosmetics.dm

```diff
@@ -82,3 +82,104 @@
 
 /obj/item/weapon/haircomb/attack_self(mob/user)
 	user.visible_message("<span class='notice'>[user] uses [src] to comb their hair with incredible style and sophistication. What a [user.gender == FEMALE ? "lady" : "guy"].</span>")
+
+/obj/item/weapon/razor
+	name = "electric razor"
+	desc = "The latest and greatest power razor born from the science of shaving."
+	icon = 'icons/obj/weapons.dmi'
+	icon_state = "razor"
+	w_class = 2
+
+/obj/item/weapon/razor/proc/shave(mob/living/carbon/human/H, location)
+	if(location == "head")
+		H.h_style = H.species.default_h_style
+	else
+		H.f_style = H.species.default_f_style
+
+	H.update_hair()
+	playsound(H, 'sound/items/welder2.ogg', 20, 1)
+
+
+/obj/item/weapon/razor/attack(mob/M, mob/user, var/target_zone)
+	if(!ishuman(M))
+		return

```

Comment:
Might want to do a return ..() to do the default attack or assign `mob/M` on line 103 to `mob/living/carbon/human/H` so it removes the redundant cast.